src/server/game/Entities/Player/Player.cpp

```diff
@@ -13667,11 +13672,11 @@ void Player::PrepareGossipMenu(WorldObject* source, uint32 menuId, bool showQues
                         canTalk = false;
                     break;
                 case GossipOptionNpc::DisableXPGain:
-                    if (HasPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN))
+                    if (HasPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN) || GetLevel() == DEFAULT_MAX_LEVEL)
```

Comment:
Should compare to MaxLevel from updatefields, not DEFAULT_MAX_LEVEL

Reply:
could i use Player::IsMaxLevel()?

Reply:
yes